### DISCUSSION TITLE
PyCharm plugin?

### DISCUSSION THREAD

**Discussion by elliotwaite** [modular/modular#87] [Ideas]
I've heard that a VS Code extension is in development. However, I'm curious if there are any plans to support a JetBrains plugin as well, since PyCharm is a popular Python IDE.

If the Modular team doesn't have plans to create a JetBrains plugin, it might be worth suggesting the idea to the JetBrains team. They have built plugins in the past for other languages like [Rust](https://plugins.jetbrains.com/plugin/8182-rust), [Swift](https://plugins.jetbrains.com/plugin/8240-swift), and [Nim](https://plugins.jetbrains.com/plugin/15128-nim). Collaborating with the JetBrains team could potentially benefit both parties. Perhaps the Modular team could reach out to the JetBrains team to explore this possibility.

**Comment by TZDak** [comment]
Sounds like a decent suggestion. I hope the Mojo development team sees it.

**Comment by yingjie52** [comment]
+1

**Comment by AlphaJin95** [comment]
+1

**Comment by NickW3101** [comment]
+1

**Comment by voathnak** [comment]
+1

**Comment by tkgpdx** [comment]
+1

**Comment by standwood** [comment]
+1

**Comment by hhstore** [comment]
+1

**Comment by tomsmierz** [comment]
+1

**Comment by williamw** [comment]
💯 would use this

**Comment by godardt** [comment]
+1

**Comment by jwhiggs77** [comment]
+1

**Comment by ksandvik** [comment]
I suspect this will be done by Jetbrains like their other language supported tools based on the forthcoming expansion of Mojo with the forthcoming compiler et rest.

**Comment by david-ragazzi** [comment]
+1

The JetBrains team already demonstrated (as soon as Mojo was released) that they are interested in making this plugin in collaboration with Modular.  However Modular closed the issue ticket due to other priority demands and said it would reopen it soon (https://github.com/modularml/mojo/issues/246). I hope it won't be long...

**Comment by ksandvik** [comment]
Yes and also as the language syntax evolves there needs to be constant updating and depending on the schedule and priorities it's up to the companies to jump in and reserve time for maintenance and updating.

**Comment by Sunling472** [comment]
+1

**Comment by elliotwaite** [comment]
There's an open YouTrack issue [here](https://youtrack.jetbrains.com/issue/PY-60990/Mojo-support) to add Mojo support to PyCharm. Feel free to upvote that issue to show support, or follow that issue to be informed of future updates from the JetBrains team.

**Comment by erfanzar** [comment]
1+

**Comment by lichtgestalter** [comment]
+1

**Comment by jmcarbo** [comment]
+1

**Comment by antonpavlovskyi** [comment]
+1

**Comment by jackhhchan** [comment]
+1

**Comment by KyoungsueKim** [comment]
+1

**Comment by Illia-Terekhov** [comment]
+1

**Comment by tristan829** [comment]
+1

**Comment by tran-huy-phuc** [comment]
+1

**Comment by saviorand** [comment]
+1

**Comment by glukicov** [comment]
+1

**Comment by k00000han** [comment]
+1

**Comment by kishmakov** [comment]
Colleagues,

I'm excited to invite you to try the [first plugin for Mojo](https://plugins.jetbrains.com/plugin/23371-mojo) support across all JetBrains IDEs.

At the moment, it provides highlighting and syntax checking. New features hopefully will be added step by step as it is still under development. Your feedback is very welcome (especially bug reports :-) ).

Thank you!

**Comment by KyoungsueKim** [comment]
Wow, that is a intriguing one. Let's dive right in and check it out.

2024년 2월 8일 (목) 오후 10:17, Kirill Shmakov ***@***.***>님이 작성:

**Comment by saleh-mir** [comment]
+1

**Comment by tran-huy-phuc** [comment]
@kishmakov Is your plugin now under JetBrains? https://plugins.jetbrains.com/plugin/23371-mojo

**Reply by kishmakov** [reply]
Hi @thphuc, yes, I transferred ownership of the plugin to JB since I don't have time to develop it.

**Reply by tran-huy-phuc** [reply]
Cool! Thanks @kishmakov
